File: kickstart/2020/C/C.cc
#include <iostream>
#include <vector>
using namespace std;

int isqrt(int n) {
	int x = n;
	int y = (x + 1) / 2;
	while (y < x) {
		x = y;
		y = (x + n / x) / 2;
	}
	return x;
}

int main() {
	int t, n;
	cin >> t;
	for (int i = 1; i <= t; ++i) {
		vector<int> array;
		int count = 0;
		cin >> n;
		for (int j = 0; j < n; ++j) {
			int num;
			cin >> num;
			array.push_back(num);
		}
		for (int m = 0; m < n; ++m) {
			int s = 0;
			for (int k = m; k < n; ++k) {
				s += array[k];
				if (s >= 0) {
					int z = isqrt(s);
					if (z * z == s) {
						count += 1;
					}
				}
			}
		}
		cout << "Case #" << i << ": " << count << endl;
	}
	return 0;
}

File: kickstart/2020/C/C.py
from functools import lru_cache
@lru_cache(maxsize=None)
def isqrt(n):
    x = n
    y = (x + 1) // 2
    while y < x:
        x = y
        y = (x + n // x) // 2
    return x

@lru_cache(maxsize=None)
def square(n):
    return n * n

T = int(input())
for t in range(1, T + 1):
    N = int(input())
    array = [int(s) for s in input().split(" ")]
    count = 0
    for i in range(N):
        s = 0
        for j in range(i, N):
            s += array[j]
            if s >= 0:
                x = isqrt(s)
                if square(x) == s:
                    count += 1
    print("Case #{}: {}".format(t, count))